arrow-libs/fx/arrow-fx-coroutines/src/commonMain/kotlin/arrow/fx/coroutines/Resource.kt

```diff
@@ -303,24 +303,23 @@ public interface ResourceScope : AutoCloseScope {
   public suspend fun <A> install(
     acquire: suspend AcquireStep.() -> A,
     release: suspend (A, ExitCase) -> Unit,
-  ): A
+  ): A = withContext(NonCancellable) {
+    acquire(AcquireStep).also { a -> onRelease { release(a, it) } }
+  }
 
   /** Composes a [release] action to a [Resource] value before binding. */
   @ResourceDSL
-  public suspend infix fun <A> Resource<A>.release(release: suspend (A) -> Unit): A {
-    val a = bind()
-    return install({ a }) { aa, _ -> release(aa) }
-  }
+  public suspend infix fun <A> Resource<A>.release(release: suspend (A) -> Unit): A =
+    bind().also { a -> onRelease { release(a) } }
 
   /** Composes a [releaseCase] action to a [Resource] value before binding. */
   @ResourceDSL
-  public suspend infix fun <A> Resource<A>.releaseCase(release: suspend (A, ExitCase) -> Unit): A {
-    val a = bind()
-    return install({ a }, release)
-  }
+  public suspend infix fun <A> Resource<A>.releaseCase(release: suspend (A, ExitCase) -> Unit): A =
+    bind().also { a -> onRelease { release(a, it) } }
```

Comment:
Do we need to be concerned about `bind()` being called in a non-cancellable context here? (It wasn't previously, either) - possibly a question for @nomisRev ?

It does mean `resource { ... } release {}` acts differently to `install({ ... }) { _, _ -> }`

Reply:
`Resource` is basically a fancy name for any function that you might call taking a `ResourceScope`, so I'd assume that either the function registers its own release (and this one here is extra for some reason?) or that the function registers some resources, but it leaves the final one for you to register yourself.
Inline usage of `resource` like that is stylistically tempting, but it feels easier to just do it imperatively.
Maybe what we actually want is for `resource` to be non-cancellable by default?